SCClassLibrary/Common/Collections/Dictionary.sc

```diff
@@ -371,6 +371,7 @@ Dictionary : Set {
 	}
 
 	== { arg that;
+		if (that === this) { ^true };
 		if(that.isKindOf(this.class).not) { ^false };
```

Comment:
The other Collection classes are doing strict `a.class != a.class` comparisons. It feels inconsistent to allow comparison of different varieties of Dictionary when we don't for anything else. Do we want to make this strict as well?